src/ray/raylet/raylet.cc

```diff
@@ -96,7 +96,11 @@ Raylet::Raylet(instrumented_io_context &main_service,
 Raylet::~Raylet() {}
 
 void Raylet::Start() {
-  RAY_CHECK_OK(RegisterGcs());
+  register_thread_ = std::thread([this]() mutable {
```

Comment:
Why do we need a new thread for this?

Reply:
because `Raylet::Star` and `AgentManager::HandleRegisterAgent` will run in the same thread, and `Raylet::Star` will block the thread to wait until `AgentManager::HandleRegisterAgent` finished, it's a dead lock

Reply:
Should a post from io context just work? I feel we don't need a thread here.

Reply:
raylet::Start already run in main_service_, so it's no use to post this wait function into main_service_
detail: https://github.com/ray-project/ray/blob/1949f359014b8be69709110772a54fc2f0c1cab8/src/ray/raylet/main.cc#L283

Reply:
@iycheng  Here I modified it again, please take a look again